app/src/views/home/memories_screen.tsx

```diff
@@ -0,0 +1,284 @@
+import { useState, useRef } from "react";
+import {
+  Animated,
+  ViewStyle,
+  Dimensions,
+  StyleSheet,
+  NativeScrollEvent,
+  NativeSyntheticEvent,
+  View,
+  TouchableOpacity,
+} from "react-native";
+import { TopBar } from "@/src/views/components/navigation/top_bar";
+import { Challenge } from "@/src/views/components/posts/challenge";
+import { ThemedView } from "@/src/views/components/theme/themed_view";
+import { ThemedText } from "@/src/views/components/theme/themed_text";
+import { useMemoriesViewModel } from "@/src/viewmodels/home/MemoriesScreenViewModel";
+
+const { width, height } = Dimensions.get("window");
+
+export default function MemoriesScreen({
+  navigation,
+  route,
+}: {
+  readonly navigation: any;
+  readonly route: any;
+}) {
+  const user = route.params?.user;
+  console.log("User in MemoriesScreen: ", user);
+  const { userIsGuest, challenges, icon } = useMemoriesViewModel(
+    user,
+    navigation,
+  );
+
+  const [showGuestPopup, setShowGuestPopup] = useState<string | null>(null);
+
+  // Animation for hiding groups
+  const scrollY = useRef(new Animated.Value(0)).current;
+  const [isCollapsed, setIsCollapsed] = useState(false);
+
+  const toggleThreshold = 100; // Distance to toggle the groups visibility
+
+  const handleScrollEnd = (e: NativeSyntheticEvent<NativeScrollEvent>) => {
+    const offsetY = e.nativeEvent.contentOffset.y;
+    const shouldCollapse = offsetY > toggleThreshold;
+
+    if (shouldCollapse !== isCollapsed) {
+      setIsCollapsed(shouldCollapse);
+      Animated.timing(scrollY, {
+        toValue: shouldCollapse ? 1 : 0,
+        duration: 220, // Reduced animation duration for snappier transitions
+        useNativeDriver: false,
+      }).start();
+    }
+  };
+
+  // Handle restricted navigation for guest users
+  const handleRestrictedAccess = (screenName: string) => {
+    if (userIsGuest) {
+      setShowGuestPopup(screenName);
+    } else {
+      navigation.navigate(screenName);
+    }
+  };
+
+  return (
+    <ThemedView style={styles.bigContainer} testID="memories-screen">
+      <TopBar
+        title="Strive"
+        leftIcon="people-outline"
+        leftAction={() => handleRestrictedAccess("Friends")}
```

Comment:
It would be more intuitive to have an "arrow-backward" icon to navigate back to HomeScreen in here